src/adaptations/device-layer/include/Weave/DeviceLayer/internal/GenericSoftwareUpdateManagerImpl.ipp

```diff
@@ -122,6 +122,8 @@ void GenericSoftwareUpdateManagerImpl<ImplClass>::PrepareBinding(intptr_t arg)
 
     self->Cleanup();
 
+    VerifyOrExit(self->mHaveServiceConnectivity, err = WEAVE_ERROR_INCORRECT_STATE);
```

Comment:
The previous code generated a log message specifically noting the lack of service connectivity.  I think that would be useful to preserve, to help developers make sense of an "Incorrect state" error in this case.

I could also imagine using the WEAVE_ERROR_NOT_CONNECTED error in this case.

Reply:
I thought about using `WEAVE_ERROR_NOT_CONNECTED`, I ultimately steered away from it because its use was around objects with actual connections.  But it does fit better here, so I'm changing the error and adding back the log message.   Thanks for the suggestion!

Reply:
Done